### ISSUE TITLE
MinimalPrimes giving incorrect results (in positive characteristic)

### ISSUE THREAD

**Issue by jchen419** [Macaulay2/M2#1025]
The following example shows an ideal J where the default minimalPrimes method and the new method in MinimalPrimes.m2 have different outputs, as well as an ideal Q which M2 (under the new method) thinks is not contained in its own radical:

```
Macaulay2, version 1.14.0.1
--loading configuration for package "FourTiTwo" from file /home/jchen646/.Macaulay2/init-FourTiTwo.m2
--loading configuration for package "Topcom" from file /home/jchen646/.Macaulay2/init-Topcom.m2
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems, LLLBases, PrimaryDecomposition,
               ReesAlgebra, TangentCone, Truncations

i1 : k = ZZ/32003

o1 = k

o1 : QuotientRing

i2 : d = 4

o2 = 4

i3 : S = k[x_0..x_d] ** k[y_0..y_d]

o3 = S

o3 : PolynomialRing

i4 : I0 = minors(2, submatrix'(matrix{{x_0..x_(d-1)},{x_1..x_d}},{d//2}))

               2
o4 = ideal (- x  + x x , - x x  + x x , - x x  + x x )
               1    0 2     1 3    0 4     2 3    1 4

o4 : Ideal of S

i5 : I = I0 + sub(I0, (vars S)_{d+1..2*d+1} | map(S^1,S^(d+1),0))

               2                                          2
o5 = ideal (- x  + x x , - x x  + x x , - x x  + x x , - y  + y y , - y y  + y y , - y y  + y y )
               1    0 2     1 3    0 4     2 3    1 4     1    0 2     1 3    0 4     2 3    1 4

o5 : Ideal of S

i6 : F = gens minors(2, transpose genericMatrix(S, d+1, 2))

o6 = | x_0y_1-x_1y_0 x_0y_2-x_2y_0 x_1y_2-x_2y_1 x_0y_3-x_3y_0 x_1y_3-x_3y_1 x_2y_3-x_3y_2 x_0y_4-x_4y_0
     --------------------------------------------------------------------------------------------------------------
     x_1y_4-x_4y_1 x_2y_4-x_4y_2 x_3y_4-x_4y_3 |

             1       10
o6 : Matrix S  <--- S

i7 : K = ker map(S/I, k[z_0..z_(binomial(d+1,2)-1)], F);

o7 : Ideal of k[z , z , z , z , z , z , z , z , z , z ]
                 0   1   2   3   4   5   6   7   8   9

i8 : J = ideal mingens sub(K, F)

             2 2        2                                                                                  2 2  
o8 = ideal (x y  - x x y  - x x y y  + x x y y  + x x y y  + x x y y  - 2x x y y  - x x y y  + x x y y  + x y  -
             1 4    0 2 4    1 3 2 4    2 3 1 4    0 4 2 4    1 4 2 3     1 4 1 4    2 4 1 3    2 4 0 4    4 1  
     --------------------------------------------------------------------------------------------------------------
      2       2                                                                   2              2                 
     x y y , x y y  - x x y y  - x x y y  + x x y y  + x x y y  - x x y y  + x x y  - x x y y , x y y  - x x y y  +
      4 0 2   1 3 4    0 2 3 4    1 3 1 4    2 3 0 4    0 4 2 3    1 4 1 3    3 4 1    3 4 0 2   1 2 4    0 2 2 4  
     --------------------------------------------------------------------------------------------------------------
                            2        2            2                   2                   2              2 2  
     x x y y  - x x y y  - x y y  + x y y  - x x y  + x x y y  + x x y  - x x y y  + x x y  - x x y y , x y  -
      1 2 2 3    1 2 1 4    2 1 3    2 0 4    1 3 2    2 3 1 2    0 4 2    1 4 1 2    2 4 1    2 4 0 2   1 3  
     --------------------------------------------------------------------------------------------------------------
          2                                                            2 2    2                             2      
     x x y  + x x y y  - x x y y  - 2x x y y  + x x y y  + x x y y  + x y  - x y y  + x x y y  - x x y y , x y y  -
      0 2 3    0 3 2 3    0 3 1 4     1 3 1 3    1 3 0 4    2 3 0 3    3 1    3 0 2    0 4 1 3    1 4 0 3   1 2 3  
     --------------------------------------------------------------------------------------------------------------
                                                      2                                               2      
     x x y y  - x x y y  + x x y y  - x x y y  + x x y  + x x y y  - x x y y , x x y y  - x x y y  + x y y  -
      0 2 1 4    1 2 1 3    1 2 0 4    1 3 1 2    2 3 1    0 4 1 2    1 4 0 2   0 1 2 3    0 1 1 4    1 0 4  
     --------------------------------------------------------------------------------------------------------------
                                           2              2        2                              2               
     x x y y  - x x y y  + x x y y  + x x y  - x x y y , x y y  - x y y  - x x y y  + x x y y  + x y y  - x x y y 
      0 2 1 3    1 3 0 2    2 3 0 1    0 4 1    1 4 0 1   0 2 3    0 1 4    0 1 1 3    0 1 0 4    1 0 3    0 2 0 3
     --------------------------------------------------------------------------------------------------------------
            2                              2                   2
     + x x y  - x x y y  - x x y y  + x x y  + x x y y  - x x y )
        0 3 1    0 3 0 2    1 3 0 1    2 3 0    0 4 0 1    1 4 0

o8 : Ideal of S

i9 : #minimalPrimes J

o9 = 1

i10 : needsPackage "MinimalPrimes"

o10 = MinimalPrimes

o10 : Package

i11 : installMinprimes()
minimalPrimes Ideal, decompose Ideal, and isPrime Ideal have been 
re-installed to use experimental code

i12 : J = ideal gens J; -- to clear cached values

i13 : #minimalPrimes J

o13 = 2

i14 : #minimalPrimes sub(J, k[gens S]) -- singly graded is OK

o14 = 1

i15 : PD = primaryDecomposition J;

i16 : Q = PD#1

                                                     2 2                2 2   2                        2   2      
o16 = ideal (x y  - x y , x y  - x y , x y  - x y , x y  - 2x x y y  + x y , x y y  - 2x x y y  + x x y , x y y  -
              3 4    4 3   2 4    4 2   2 3    3 2   1 4     1 4 1 4    4 1   1 3 4     1 4 1 3    3 4 1   1 2 4  
      -------------------------------------------------------------------------------------------------------------
                       2   2 2                2 2   2                        2   2 2                2 2       2  
      2x x y y  + x x y , x y  - 2x x y y  + x y , x y y  - 2x x y y  + x x y , x y  - 2x x y y  + x y , x x y  -
        1 4 1 2    2 4 1   1 3     1 3 1 3    3 1   1 2 3     1 3 1 2    2 3 1   1 2     1 2 1 2    2 1   0 1 4  
      -------------------------------------------------------------------------------------------------------------
                             2                                                                                 
      x x y y  - x x y y  + x y y , x x y y  - x x y y  - x x y y  + x x y y , x x y y  - x x y y  - x x y y  +
       0 4 1 4    1 4 0 4    4 0 1   0 1 3 4    0 4 1 3    1 4 0 3    3 4 0 1   0 1 2 4    0 4 1 2    1 4 0 2  
      -------------------------------------------------------------------------------------------------------------
                            2            2                  2                          2                           
      x x y y , x x y y  - x y y  - x x y  + x x y y , x x y  - x x y y  - x x y y  + x y y , x x y y  - x x y y  -
       2 4 0 1   0 1 1 4    1 0 4    0 4 1    1 4 0 1   0 1 3    0 3 1 3    1 3 0 3    3 0 1   0 1 2 3    0 3 1 2  
      -------------------------------------------------------------------------------------------------------------
                                       2            2                  2                          2                
      x x y y  + x x y y , x x y y  - x y y  - x x y  + x x y y , x x y  - x x y y  - x x y y  + x y y , x x y y  -
       1 3 0 2    2 3 0 1   0 1 1 3    1 0 3    0 3 1    1 3 0 1   0 1 2    0 2 1 2    1 2 0 2    2 0 1   0 1 1 2  
      -------------------------------------------------------------------------------------------------------------
       2            2              2 2                2 2   2                        2   2                  
      x y y  - x x y  + x x y y , x y  - 2x x y y  + x y , x y y  - 2x x y y  + x x y , x y y  - 2x x y y  +
       1 0 2    0 2 1    1 2 0 1   0 4     0 4 0 4    4 0   0 3 4     0 4 0 3    3 4 0   0 2 4     0 4 0 2  
      -------------------------------------------------------------------------------------------------------------
           2   2                                  2   2 2                2 2   2                        2   2      
      x x y , x y y  - x x y y  - x x y y  + x x y , x y  - 2x x y y  + x y , x y y  - 2x x y y  + x x y , x y y  -
       2 4 0   0 1 4    0 1 0 4    0 4 0 1    1 4 0   0 3     0 3 0 3    3 0   0 2 3     0 3 0 2    2 3 0   0 1 3  
      -------------------------------------------------------------------------------------------------------------
                                 2   2 2                2 2   2                                  2   2 2  
      x x y y  - x x y y  + x x y , x y  - 2x x y y  + x y , x y y  - x x y y  - x x y y  + x x y , x y  -
       0 1 0 3    0 3 0 1    1 3 0   0 2     0 2 0 2    2 0   0 1 2    0 1 0 2    0 2 0 1    1 2 0   0 1  
      -------------------------------------------------------------------------------------------------------------
                   2 2
      2x x y y  + x y )
        0 1 0 1    1 0

o16 : Ideal of S

i17 : isSubset(Q, radical Q)

o17 = false
```
Note that the ambient ring S is bigraded. Also, these examples do not seem to fail in characteristic 0.

**Comment by DanGrayson** [comment]
Is the bug in the Core or in the package?

**Comment by jchen419** [comment]
Hi Dan, I believe the bug is in the package MinimalPrimes, not Core. (At least mathematically, I believe the ideal J is irreducible, and certainly the result about Q and its radical -- obtained using the package -- is wrong.)

**Comment by mikestillman** [comment]
I've just assigned myself to this one.

**Comment by mahrud** [comment]
Fixed in #1385
